packages/next/src/server-build.ts

```diff
@@ -2279,7 +2279,11 @@ export async function serverBuild({
             prefetchSegmentSuffix
               ? [
                   {
-                    src: path.posix.join('/', entryDirectory, '/(?<path>.+)$'),
+                    src: path.posix.join(
+                      '/',
+                      entryDirectory,
+                      '/(?<path>.+?)(?:/)?$'
```

Comment:
Will defer to @ztanner / @ijjk. One thing I'm wondering is if we should add this optional match for trailing `/` or instead we should know when trailingSlash is configured to include/exclude it.